### PR TITLE
Add support for URLs in docstring (#108)

### LINKED ISSUE #108: Support URLs in docstring
Defopt seems to raise an error if you have a URL (both plain text URLs and [RST-formatted URLs](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages)) in your docstring. I can submit a PR to fix this.

### PR DESCRIPTION
**Pull request by neelmraman**
closes #108

### CODE REVIEW

**Comment by anntzer** [comment]
```rst
Go to Google_.

.. _Google: https://www.google.com
```
currently breaks too, can you fix that as well? I think you just need to Skip the target node when current_paragraph is None?  (Then the URI doesn't show up in the rendered text, which seems correct.)

File: lib/defopt.py [hunk 1]
```diff
@@ -754,6 +754,15 @@ def visit_Text(self, node):
 
         depart_Text = _do_nothing
 
+        visit_reference = depart_reference = _do_nothing
+
```
Comment (anntzer) [inline lib/defopt.py:758]: you have some trailing whitespace here.

File: test_defopt.py [hunk 2]
```diff
@@ -1163,6 +1163,24 @@ def foo(bar):
...
 
+    def test_hyperlink_plaintext(self):
+        def foo():
+            """This site https://www.python.org/ is cool"""
+        self.assert_in_help(
+            "This site https://www.python.org/ is cool",
+            foo,
+            ""
+        )
+
```
Comment (anntzer) [inline test_defopt.py:1174]: and trailing whitespace here too

**Comment by neelmraman** [comment]
> ```
> Go to Google_.
> 
> .. _Google: https://www.google.com
> ```
> 
> currently breaks too, can you fix that as well? I think you just need to Skip the target node when current_paragraph is None? (Then the URI doesn't show up in the rendered text, which seems correct.)

Ah yeah I didn't think the non-embedded form was really used in docstrings but I'll add support for it

**Comment by anntzer** [comment]
Thanks!

**Pull request merged**